x-pack/plugins/telemetry_collection_xpack/server/plugin.ts

```diff
@@ -5,27 +5,27 @@
  */
 
 import { PluginInitializerContext, CoreSetup, CoreStart, Plugin } from 'kibana/server';
-import { TelemetryCollectionManagerPluginSetup } from 'src/plugins/telemetry_collection_manager/server';
+import { TelemetryCollectionManagerPluginStart } from 'src/plugins/telemetry_collection_manager/server';
 import { getClusterUuids, getLocalLicense } from '../../../../src/plugins/telemetry/server';
 import { getStatsWithXpack } from './telemetry_collection';
 
-interface TelemetryCollectionXpackDepsSetup {
-  telemetryCollectionManager: TelemetryCollectionManagerPluginSetup;
+interface TelemetryCollectionXpackDepsStart {
+  telemetryCollectionManager: TelemetryCollectionManagerPluginStart;
 }
 
 export class TelemetryCollectionXpackPlugin implements Plugin {
   constructor(initializerContext: PluginInitializerContext) {}
 
-  public setup(core: CoreSetup, { telemetryCollectionManager }: TelemetryCollectionXpackDepsSetup) {
+  public setup(core: CoreSetup) {}
+
+  public start(core: CoreStart, { telemetryCollectionManager }: TelemetryCollectionXpackDepsStart) {
     telemetryCollectionManager.setCollection({
-      esCluster: core.elasticsearch.dataClient,
+      esCluster: core.elasticsearch.legacy.client,
```

Comment:
@afharo do you know a reason why it couldn't work if we move `setCollection` to the start lifecycle?

Reply:
I bet there's a point in the code where we are trying to read the telemetry from the `setup` lifecycle step (on top of my mind https://github.com/elastic/kibana/blob/dacc95fa85788bf0a3cd77074159b7512b6f61fb/x-pack/plugins/monitoring/server/plugin.ts#L227 but there might be more).

I can try to dig deeper on it :)

Reply:
Ha. Good to know. I will try to move this piece of the code in `start` lifecycle. 

> I can try to dig deeper on it :)

I will ping you if I didn't fix it with ^^^